libs/state/src/lib/rx-state.service.ts

```diff
@@ -405,64 +405,104 @@ export class RxState<T extends object> implements OnDestroy, Subscribable<T> {
       | Signal<T[K] | V>,
     projectValueFn?: ProjectValueReducer<T, K, V>
   ): void {
-    let inputOrSlice$: Observable<Partial<T> | V> | undefined;
-    if (!isKeyOf<T>(keyOrInputOrSlice$)) {
-      if (isObservable(keyOrInputOrSlice$)) {
-        inputOrSlice$ = keyOrInputOrSlice$;
-      } else {
-        // why can't typescript infer the correct type?
-        inputOrSlice$ = toObservable(
-          keyOrInputOrSlice$ as Signal<Partial<T> | V>,
-          { injector: this.injector }
-        );
-      }
+    /**
+     * From top to bottom the overloads are handled.
+     */
+    if (
+      isObservable(keyOrInputOrSlice$) &&
+      !projectOrSlices$ &&
+      !projectValueFn
+    ) {
+      this.accumulator.nextSliceObservable(keyOrInputOrSlice$);
+      return;
+    }
+
+    if (isSignal(keyOrInputOrSlice$) && !projectOrSlices$ && !projectValueFn) {
+      this.accumulator.nextSliceObservable(
+        toObservable(keyOrInputOrSlice$, { injector: this.injector })
+      );
+      return;
     }
-    const key: K | null =
-      !inputOrSlice$ && isKeyOf<T>(keyOrInputOrSlice$)
-        ? keyOrInputOrSlice$
-        : null;
+
     if (
-      projectValueFn === undefined &&
-      projectOrSlices$ === undefined &&
-      inputOrSlice$
+      isObservable(keyOrInputOrSlice$) &&
+      projectOrSlices$ &&
+      typeof projectOrSlices$ === 'function' &&
+      !projectValueFn
     ) {
-      this.accumulator.nextSliceObservable(inputOrSlice$);
+      const projectionStateFn = projectOrSlices$;
+      const slice$ = keyOrInputOrSlice$.pipe(
+        map((v) => projectionStateFn(this.accumulator.state, v as V))
+      );
+      this.accumulator.nextSliceObservable(slice$);
```

Comment:
Type error could be fixed like so: 

```suggestion
      this.accumulator.nextSliceObservable(slice$ as Observable<V>);
```